src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationWriter.cs

```diff
@@ -38,6 +38,9 @@ public abstract class XmlSerializationWriter : XmlSerializationGeneratedCode
         private bool _soap12;
         private bool _escapeName = true;
 
+        //char buffer for serializing primitive values
+        private char[]? _primitivesBuffer;
```

Comment:
As @mconnew mentioned, it should be safe to keep one buffer alongside the reference to a single instance of XmlWriter (aka, _w on line 27). Calling into here simultaneously would already be asking for trouble racing to the `_w.Write*()` calls at the end of the method.

```suggestion
        private char[] _primitivesBuffer = new char[128];
```

And then use this buffer through the rest of `WriteTypedPrimitive` without the need for any interlocked exchange.

Reply:
@StephenMolloy, @mconnew
I will remove it, but have two questions:
 - 128 might be too big - Do you see any primitive type that cannot fit in 64 for example?
 - If you check [next changes](https://github.com/dotnet/runtime/pull/81687), we will need same buffer in other classes that inherits it. Should we make it protected with this PR or later?

Reply:
I have following stopper if we cannot format some value in this buffer:
![image](https://user-images.githubusercontent.com/12857389/227513898-13f9435b-5723-440d-a05f-25bb13474819.png)

Maybe this is enough to experiment with 64, or if you prefer I can replace Debug.Assert with throw exception

Reply:
Okay - I set it to char[64]